src/hotspot/share/opto/loopnode.cpp

```diff
@@ -4311,7 +4311,7 @@ void PhaseIdealLoop::mark_loop_associated_parse_predicates_useful() {
 }
 
 void PhaseIdealLoop::mark_useful_parse_predicates_for_loop(IdealLoopTree* loop) {
-  Node* entry = loop->_head->in(LoopNode::EntryControl);
+  Node* entry = loop->_head->as_Loop()->skip_strip_mined()->in(LoopNode::EntryControl);
```

Comment:
Is now required due to changing `can_apply_loop_predication()` which is checked before calling this method.